src/AbstractContainer.php

```diff
@@ -477,11 +473,10 @@ public function exchangeArray($input)
 
         $old            = $storage[$name];
         $storage[$name] = $input;
-        if ($old instanceof ArrayObject) {
-            return $old->getArrayCopy();
-        }
 
-        return $old;
+        /** @var array<TKey, TValue> $return */
+        $return = $old instanceof ArrayObject ? $old->getArrayCopy() : $old;
+        return $return;
```

Comment:
You can avoid temporary variables in psalm with

```suggestion
        /** @psalm-var array<TKey, TValue> */
        return $old instanceof ArrayObject ? $old->getArrayCopy() : $old;
```